src/pods/generate/TemplatedPodGenerator.ts

```diff
@@ -0,0 +1,73 @@
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { getLoggerFor } from '../../logging/LogUtil';
+import type { KeyValueStorage } from '../../storage/keyvalue/KeyValueStorage';
+import type { ResourceStore } from '../../storage/ResourceStore';
+import { BadRequestHttpError } from '../../util/errors/BadRequestHttpError';
+import { ConflictHttpError } from '../../util/errors/ConflictHttpError';
+import { joinFilePath } from '../../util/PathUtil';
+import type { TemplateAgent } from '../agent/TemplateAgent';
+import type { ComponentsJsFactory } from './ComponentsJsFactory';
+import type { PodGenerator } from './PodGenerator';
+import type { VariableHandler } from './variables/VariableHandler';
+import { TEMPLATE_STORE_URN, TEMPLATE_CONFIG_URN, isValidVariable } from './variables/Variables';
+
+const DEFAULT_CONFIG_PATH = joinFilePath(__dirname, '../../../config/templates');
+
+/**
+ * Creates a new ResourceStore when creating a pod based on a Components.js configuration.
+ */
+export class TemplatedPodGenerator implements PodGenerator {
+  protected readonly logger = getLoggerFor(this);
+  private readonly storeFactory: ComponentsJsFactory;
+  private readonly variableHandler: VariableHandler;
+  private readonly configStorage: KeyValueStorage<string, unknown>;
+  private readonly configTemplatePath: string;
+
+  /**
+   * @param storeFactory - Factory used for Components.js instantiation.
+   * @param variableHandler - Handler used for setting variable values.
+   * @param configStorage - Where to store the configuration values to instantiate the store for this pod.
+   * @param configTemplatePath - Where to find the configuration templates.
+   */
+  public constructor(storeFactory: ComponentsJsFactory, variableHandler: VariableHandler,
+    configStorage: KeyValueStorage<string, unknown>, configTemplatePath?: string) {
+    this.storeFactory = storeFactory;
+    this.variableHandler = variableHandler;
+    this.configStorage = configStorage;
+    this.configTemplatePath = configTemplatePath ?? DEFAULT_CONFIG_PATH;
+  }
+
+  public async generate(identifier: ResourceIdentifier, agent: TemplateAgent): Promise<ResourceStore> {
+    if (await this.configStorage.has(identifier.path)) {
+      this.logger.warn(`There already is a pod at ${identifier.path}`);
+      throw new ConflictHttpError(`There already is a pod at ${identifier.path}`);
+    }
+
+    await this.variableHandler.handleSafe({ identifier, agent });
+
+    // Filter out irrelevant data in the agent
+    const variables: NodeJS.Dict<string> = {};
+    for (const key of Object.keys(agent)) {
+      if (isValidVariable(key)) {
+        variables[key] = agent[key];
+      }
+    }
+
+    // Prevent unsafe template names
+    if (!/^[a-zA-Z0-9.-]+$/u.test(agent.template)) {
```

Comment:
Or perhaps as a function `validatePodName` etc.

Reply:
The one in slug is currently slightly more strict and also slightly different since it matches single characters so it can replace them. But I do like the idea of grouping them somewhere.

Reply:
Thinking about it again I'm going to leave this as it is since the regex here is sufficiently different from the one for validating a pod name, and I don't see that much value in adding unrelated regexes to a single file (until some of them get re-used of course).